api/data_refinery_api/management/commands/post_downloads_summary.py

```diff
@@ -39,8 +39,19 @@ def handle(self, *args, **options):
             if annotation.data["start"] and should_display_email(annotation.dataset.email_address)
         ]
 
+        # Save the locations permanently, since IP addresses can cycle over time
+        location_cache = dict()
+        for annotation in annotation_queryset:
+            if not "location" in annotation.data:
+                ip = annotation.data["ip"]
+                print(ip)
+                if not ip in location_cache:
+                    location_cache[ip] = get_ip_location(ip)
+                annotation.data["location"] = location_cache[ip]
+                annotation.save()
```

Comment:
Why wait until we post a summary to do this? It seems like we could add these annotations when the dataset is created, then just report on them from this command.

Reply:
`get_ip_location` calls an external API to get the location of each ip address, and generally I think people try to avoid calling external APIs before their own API returns because it causes extra request latency.

Reply:
Ah yes. That's a very good point!